README.md

```diff
@@ -1,2 +1,209 @@
+
+You can interact with the service using any HTTP client. Here's a Python example using the `requests` library:
+
+```python
+import requests
+
+# Initialize base URL
+base_url = "http://localhost:8005"
+
+# Get embedding for a query
+response = requests.post(
+    f"{base_url}/api/v1/embed/query",
+    json={"text": "What is copyright infringement?"}
+)
+query_embedding = response.json()["embedding"]
+
+# Get embeddings for a document
+response = requests.post(
+    f"{base_url}/api/v1/embed/text",
+    data="The court finds that..."
+)
+doc_embeddings = response.json()["embeddings"]
+```
+
+For more examples and detailed API documentation, check the API endpoints section below.
+
+## API Endpoints
+
+### Query Embeddings
+Generate embeddings for search queries (CPU-optimized):
+```bash
+curl 'http://localhost:8005/api/v1/embed/query' \
+  -X 'POST' \
+  -H 'Content-Type: application/json' \
+  -d '{"text": "What are the requirements for copyright infringement?"}'
+```
+
+### Document Embeddings
+Generate embeddings for court opinions or legal documents (GPU-accelerated when available):
+```bash
+curl 'http://localhost:8005/api/v1/embed/text' \
+  -X 'POST' \
+  -H 'Content-Type: text/plain' \
+  -d 'The court finds that the defendant...'
+```
+
+### Batch Processing
+Process multiple documents in one request:
+```bash
+curl 'http://localhost:8005/api/v1/embed/batch' \
+  -X 'POST' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "documents": [
+      {"id": 1, "text": "First court opinion..."},
+      {"id": 2, "text": "Second court opinion..."}
+    ]
+  }'
+```
+
+## Configuration
+
+The service can be configured through environment variables or a `.env` file. Copy `.env.example` to `.env` to get started:
+```bash
+cp .env.example .env
+```
+
+### Environment Variables
+
+Model Settings:
+- `TRANSFORMER_MODEL_NAME`: Model to use (default: "sentence-transformers/multi-qa-mpnet-base-dot-v1")
+- `MAX_WORDS`: Maximum words per chunk (default: 350)
+
+Server Settings:
+- `HOST`: Server host (default: "0.0.0.0")
+- `PORT`: Server port (default: 8005)
+- `EMBEDDING_WORKERS`: Number of Gunicorn workers (default: 4)
+
+GPU Settings:
+- `FORCE_CPU`: Force CPU usage even if GPU is available (default: false)
+
+Monitoring:
+- `SENTRY_DSN`: Sentry DSN for error tracking (optional)
+- `ENABLE_METRICS`: Enable Prometheus metrics (default: true)
+
+CORS Settings:
+- `ALLOWED_ORIGINS`: Comma-separated list of allowed origins
+- `ALLOWED_METHODS`: Comma-separated list of allowed methods
+- `ALLOWED_HEADERS`: Comma-separated list of allowed headers
+
+See `.env.example` for a complete list of configuration options.
+
+## Development and Testing
+
+For development setup and testing instructions, see [DEVELOPING.md](DEVELOPING.md).
+
+## Contributing
+
+We welcome contributions to improve the embedding service!
+
+1. For development setup, see [DEVELOPING.md](DEVELOPING.md)
+2. For submitting changes, see [SUBMITTING.md](SUBMITTING.md)
```

Comment:
Do these files exist?

Reply:
Oh no, they don't exist! I thought they were added to explain those two concepts in more detail in the future. But yeah, it seems better to remove now them from here if we don't have enough content to create those separate files.

Currently, the README includes basic steps for running the project and tests, which cover development. I've also added a couple of points to the "Contributing" section.

Let me know if there's anything else we should add to improve these sections.